### ISSUE TITLE
 brew cask upgrade fails with "No Cask with this name exists"

### ISSUE THREAD

**Issue by mpdehnel** [Homebrew/homebrew-cask#101136]
### 

- [X] I understand that [if I ignore these instructions, my issue may be closed without review](https://github.com/Homebrew/homebrew-cask/blob/master/doc/faq/closing_issues_without_review.md).
- [X] I have retried my command with `--force`.
- [X] I ran `brew update-reset && brew update` and retried my command.
- [X] I ran `brew doctor`, fixed as many issues as possible and retried my command.
- [X] I have checked the instructions for [reporting bugs](https://github.com/Homebrew/homebrew-cask#reporting-bugs).
- [X] I made doubly sure this is not a [checksum does not match](https://github.com/Homebrew/homebrew-cask/blob/master/doc/reporting_bugs/checksum_does_not_match_error.md) error.

### Description of issue

So, identical to the issue described in [this issue](https://github.com/Homebrew/homebrew-cask/issues/72562#issuecomment-568637425) I've had my whole brew setup mostly made unusable by a series of old casks that have since been removed. I have had to manually remove the contents of the old casks from my file system, as not doing so left me with `brew` completely unusable.
I was trying to `brew untap homebrew/homebrew-science --force` but this just got blocked on old casks, e.g.
`Error: Cask 'mailbox' is unavailable: No Cask with this name exists.`
Come on! As you you say in the other issue (linked above) this is "suboptimal" (seriously?!) — this screwed over my whole homebrew install.

`brew update-reset && brew update` fails because it gets stuck on the old homebrew-science cask, which I was trying to remove...:
```
==> Fetching /usr/local/Homebrew/Library/Taps/homebrew/homebrew-science...
git: 'credential-gcloud.sh' is not a git command. See 'git --help'.
Username for 'https://github.com':
```

Any attempts at `brew doctor` then fail, being blocked by e.g.
```
Error: Cask 'mailbox' is unavailable: No Cask with this name exists.
```
and 
```
brew untap homebrew/homebrew-science --force
Error: Cask 'seil' is unavailable: No Cask with this name exists.
```
...but I had to do this about 10 times.

If a cask doesn't exist, you can't just block all use of homebrew! :cry:

### Command that failed

All of them

### Output of command with `--verbose --debug`

```
brew untap homebrew/homebrew-science --verbose --debug
Warning: Unexpected method 'accessibility_access' called on Cask alfred.
Follow the instructions here:
  https://github.com/Homebrew/homebrew-cask#reporting-bugs
Error: Cask 'appcleaner' definition is invalid: invalid 'depends_on macos' value: unknown or unsupported macOS version: :tiger
```

### Output of `brew doctor --verbose`
```
Error: Cask 'mailbox' is unavailable: No Cask with this name exists.
```

### Output of `brew tap`
```
brewsci/science
homebrew/cask
homebrew/cask-fonts
homebrew/core
teamookla/speedtest
```

**Comment by vitorgalvao** [comment]
> Come on! As you you say in the other issue (linked above) this is "suboptimal" (seriously?!)

If you’ve opened an issue just to vent, I’d recommend you open a PR instead. You’re welcome to use Homebrew [without any guarantees, expressed or implied](https://github.com/Homebrew/brew/blob/master/LICENSE.txt), but not to demand or expect free work from our volunteer efforts.

> If a cask doesn't exist, you can't just block all use of homebrew!

Yes, we know it’s frustrating; we don’t want this behaviour either and [there’s already another issue open for this](https://github.com/Homebrew/homebrew-cask/issues/71221). Please search before opening issues; there aren’t that many.

**Comment by mpdehnel** [comment]
Hey.

I'm sorry for having a bit of a vent. Long day, and frustrating when stuff like this breaks badly :-(
I also really appreciate your efforts for creating, working on, and maintaining this awesome software — I don't know what I'd do without Homebrew! Makes it all the sadder when it breaks.

The reason I didn't find the open issue you just linked to was because when I googled the particular error messages, all the top suggestions that came up were like the one I linked to above — closed — and without any indication of the current open issue, or suggested workarounds or fixes. If it had been linked to in the issues I found — where you replied closing it — I would have realised there was no point in opening a new issue!

If you would be able to post a recommended workaround, resolution, or temporary fix (no matter how ugly) in closing issues like this, it will probably help people who discover it through google in future again! Really appreciate it.

Thank you :-)

Martin
